### PR TITLE
feat(zone): abstract heap, virtual, and durable object interface

### PR DESCRIPTION
**Pull request by michaelfig**
## Description

This is an MVP for the Zones API as conceived by @erights and fleshed out for an immediate use (#7107).  The API is not yet considered stable, so it may still evolve.  Please do not use it outside the SDK yet.

### Security Considerations

n/a

### Scaling Considerations

n/a

### Documentation Considerations

We'll need to document Zones sometime, probably first in code, then using that as source material for the documentation website.

### Testing Considerations

Most of this API is just a thin wrapper for `@agoric/vat-data` and `@agoric/store`, so the implementation is tested in those packages.  Also, #7107 exercises a lot of the API and passes tests while using it.

### CODE REVIEW

File: packages/zone/src/durable.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  const attachedStores = attachDurableStores(() => baggage);
+
+  /** @type {import('.').Zone['subZone']} */
+  const subZone = (label, options = {}) => {
+    const subBaggage = provideDurableMapStore(baggage, label, options);
+    return makeDurableZone(subBaggage);
+  };
+
+  return harden({
+    exoSingleton,
```
Comment (erights) [inline packages/zone/src/durable.js]:
```suggestion
    exo,
```
please

File: packages/zone/src/index.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  weakSetStore: makeScalarWeakSetStore,
+});
+
+/**
+ * An ephemeral (in-memory) zone that uses the default exo and store
+ * implementations.
+ *
+ * @type {import('.').Zone}
+ */
+export const ephemeralZone = harden({
```
Comment (erights) [inline packages/zone/src/index.js]: Should zones be remotable?

**Review by erights — COMMENTED** [review]
LGTMSF -- looks good to me so far!

Solved my biggest "Place" problem elegantly.

Please do try the ERTP experiment.

File: packages/zone/src/durable.js (hunk 1 above)
Comment (michaelfig) [reply packages/zone/src/durable.js]: Ah, yes, forgot about this one!  Done.

File: packages/zone/src/index.js (hunk 2 above)
Comment (michaelfig) [reply packages/zone/src/index.js]: That's a good idea.  Also done.

**Comment by erights** [comment]
> Please do try the ERTP experiment.

In progress at https://github.com/Agoric/agoric-sdk/pull/7116

File: packages/zone/src/index.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { makeExo, defineExoClass, defineExoClassKit } from '@agoric/store';
+
+export * from './heap.js';
+
+// References to allow the below typeofs to succeed.
+makeExo;
+defineExoClass;
+defineExoClassKit;
```
Comment (erights) [inline packages/zone/src/index.js:8]: TIL

**Review by erights — APPROVED** [review]
LGTM. Forward!

**Pull request merged**